torch_frame/data/multi_tensor.py

```diff
@@ -325,6 +324,43 @@ def select(
     def _single_index_select(self, index: int, dim: int) -> _MultiTensor:
         raise NotImplementedError
 
+    def fillna_(self, fill_value: Union[int, float, Tensor]):
+        """Fill the :obj:`MultiTensor` with fill_value.
+
+        Args:
+            index (int): A column index of the tensor to select.
+            fill_value (Union[int, float, Tensor]): Scalars or vector
+                tensor of size (num_cols,) to replace NaNs.
+        """
+        if isinstance(fill_value, Tensor):
+            if fill_value.dim() != 1 or fill_value.numel() != self.num_cols:
+                raise ValueError(
+                    "fillna_ with Tensor of shape "
+                    f"{fill_value.shape} is not supported. "
+                    f"Please use a Tensor of shape ({self.num_cols}.)")
+            for col in range(self.num_cols):
+                self.fillna_col(col, fill_value[col])
+            return
```

Comment:
From what I see, there is not much efficiency gain with this since we are doing for-loop over columns anyway. I'd just delete fill_value is Tensor case.

Reply:
Sure.